server/service/datadeal/validate_data_handler_test.go

```diff
@@ -0,0 +1,246 @@
+	return mux.SetURLVars(req, pathMap)
+}
+
+func TestHandlerValidateData(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, bytes.NewReader(defaultData))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	res := recorder.Result()
+
+	require.Equal(t, http.StatusCreated, res.StatusCode)
+	resBody := recorder.Body.Bytes()
+
+	cert := markettypes.DataCert{}
+	err := codec.ProtoUnmarshalJSON(resBody, &cert)
+	require.NoError(t, err)
+	unsignedCert := cert.UnsignedCert
+	require.Equal(t, uint64(1), unsignedCert.DealId)
+	// check if dataUrl is encrypted with buyer's pubKey
+	buyerPrivKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), buyerPrivKey.Bytes())
+	url, err := btcec.Decrypt(buyerPrivKey, unsignedCert.EncryptedDataUrl)
+	require.NoError(t, err)
+	require.True(t, strings.HasPrefix(string(url), "1/"))
+	require.Equal(t, oracleAccount.GetAddress(), unsignedCert.OracleAddress)
+	require.Equal(t, crypto.Hash(defaultData), unsignedCert.DataHash)
+	require.Equal(t, requesterAddress, unsignedCert.RequesterAddress)
+	// verify that oracle signed it
+	serializedCertificate, err := unsignedCert.Marshal()
+	require.NoError(t, err)
+	require.True(t, oracleAccount.GetSecp256k1PubKey().VerifySignature(serializedCertificate, cert.Signature))
+}
+
+func TestHandlerValidateDataInvalidContentType(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, bytes.NewReader(defaultData))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/text")
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	require.Equal(t, http.StatusUnsupportedMediaType, recorder.Result().StatusCode)
+	require.Equal(t, "only application/json is supported\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataInvalidParameter(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data", bytes.NewReader(defaultData))
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+
+	recorder := httptest.NewRecorder()
+
+	dealService := makeDataDealService()
+
+	dealService.handleValidateData(recorder, req)
+
+	require.Equal(t, http.StatusBadRequest, recorder.Result().StatusCode)
+	require.Equal(t, "failed to read query parameter\n", recorder.Body.String())
+}
+
+func TestHandlerValidateDataEmptyBody(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/v0/data-deal/deals/1/data?requester_address="+requesterAddress, nil)
+	req = setDefaultURLVars(req)
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Form = make(map[string][]string)
+	req.Form.Set("requester_address", requesterAddress)
```

Comment:
When you make a new request, the requesterAddress is given to the request, but it seems to be given again here.